plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -111,6 +112,32 @@ typedef struct __attribute__((packed)) {
 	guint8 reserved[22];
 } FuGenesysDynamicToolString;
 
+typedef struct __attribute__((packed)) {
+	guint8 running_mode; /* 'M' for mask code, the others for bank code */
+
+	guint8 ss_port_number; /* super-speed port number */
+	guint8 hs_port_number; /* high-speed port number */
+
+	guint8 ss_connection_status; /* bit field. ON = DFP is a super-speed device */
+	guint8 hs_connection_status; /* bit field. ON = DFP is a high-speed device */
+	guint8 fs_connection_status; /* bit field. ON = DFP is a full-speed device */
+	guint8 ls_connection_status; /* bit field. ON = DFP is a low-speed device */
+
+	guint8 charging;		  /* bit field. ON = DFP is a charging port */
+	guint8 non_removable_port_status; /* bit field. ON = DFP is a non-removable port */
+	guint8 bonding;
+
+	// indicate current FW runs on which bank, 0 means disable, 1 means bank 0, 2 means bank 1.
+	// also record PD and Dev FW version.
+	guint8 hub_fw_status;
+	guint8 pd_fw_status;
+	guint8 pd_fw_version_low;
+	guint8 pd_fw_version_high;
+	guint8 dev_fw_status;
+	guint8 dev_fw_version_low;
+	guint8 dev_fw_version_high;
+} FuGenesysDynamicToolString_GL3525;
```

Comment:
this is a really good candidate for moving to fu-genesys.rs -- i.e. see https://github.com/fwupd/fwupd/blob/main/docs/tutorial.md#the-rustgen-helper

Reply:
I'm going to try it.

Reply:
@hughsie Is there any formating for `.rs`? e.g. `clang-foramt`

Reply:
I committed the code to extend some attributes. Then I will try to convert to use `.rs` now. If there are any suggestions in these commits, please kindly point out.

Reply:
rustfmt can help a lot -- although the generator also verifies syntax too.